spec/requests/restaurants_spec.rb

```diff
@@ -0,0 +1,132 @@
+require 'rails_helper'
+
+RSpec.describe 'Restaurants', type: :request do
+  before(:all) do
+    FactoryBot.create_list(:restaurant, 10)
+  end
+
+  describe 'GET /restaurants' do
+    before do
+      get restaurants_path
+    end
+
+    it 'return https success' do
+      expect(response).to have_http_status(200)
+    end
+
+    it 'has the right objects count' do
+      arr = response_body
+      expect(arr.count).to eq(Restaurant.all.count)
+    end
+
+    it 'contains mandatory attributes' do
+      arr = response_body
+      arr.each do |json_rest|
+        expect(json_rest.keys).to include('id', 'name', 'cuisine', 'accept10bis')
```

Comment:
also, it won't work here but just so you will be familiar with https://simpleror.wordpress.com/2009/03/15/q-q-w-w-x-r-s/